libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/LongPressMapComponent.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.dropin.component.marker
+
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.gestures.OnMapLongClickListener
+import com.mapbox.maps.plugin.gestures.gestures
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.DropInNavigationViewModel
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.dropin.model.Destination
+
+internal class LongPressMapComponent(
+    private val mapView: MapView,
+    private val viewModel: DropInNavigationViewModel
```

Comment:
Can you remove the top level `DropInNavigationViewModel`

Would like to have the destination logic be part of a UIViewModel

Reply:
Which UIViewModel?

Reply:
It will depend on whether we want to store the destination across orientation changes. It personally makes sense to me store the destinations and waypoints. In that case we can create a `DestinationViewModel : UIViewModel<>` that would keep track of that data. Hence, when we create a `GeocodingComponent` that could depend on `DestinationViewModel`. If it seems too early to create this view model, we can store the destination data in `RouteViewModel`